### PR TITLE
[Docs] Improvements to JSON-based batch Ingestion page

### PR DESCRIPTION
**Pull request by ektravel**
This PR:

- Replaces "native ingestion" with "JSON-based batch ingestion"
- Adds missing code font
- Capitalizes column names
- Fixes a broken table
- Removes some of the passive voice and future tense
- Encloses long code samples in the `<details>` tag to create interactive widgets

This PR has:

- [x] been self-reviewed.

### CODE REVIEW

File: docs/ingestion/native-batch.md [hunk 1]
```diff
@@ -188,65 +211,70 @@ The following example illustrates the configuration for a parallel indexing task
...
+|`type`|The task type. Set to the value to `index_parallel`.|none|yes|
+|`inputFormat`|[`inputFormat`](./data-formats.md#input-format) to specify how to parse input data.|none|yes|
+|`appendToExisting`|Creates segments as additional shards of the latest version, effectively appending to the segment set instead of replacing it. This means that you can append new segments to any datasource regardless of its original partitioning scheme. You must use the `dynamic` partitioning type for the appended segments. If you specify a different partitioning type, the task fails with an error.|false|no|
+|`dropExisting`|If `true` and `appendToExisting` is `false` and the `granularitySpec` contains an`interval`, then the ingestion task replaces all existing segments fully contained by the specified `interval` when the task publishes new segments. If ingestion fails, Druid doesn't change any existing segments. In the case of misconfiguration where either `appendToExisting` is `true` or `interval` isn't specified in `granularitySpec`, Druid doesn't replace any segments even if `dropExisting` is `true`. WARNING: this feature is still in beta.|false|no|
```
Comment (317brian) [inline docs/ingestion/native-batch.md:245]: I think technically there's only an experimental stage, not beta

```suggestion
|`dropExisting`|If `true` and `appendToExisting` is `false` and the `granularitySpec` contains an`interval`, then the ingestion task replaces all existing segments fully contained by the specified `interval` when the task publishes new segments. If ingestion fails, Druid doesn't change any existing segments. In the case of misconfiguration where either `appendToExisting` is `true` or `interval` isn't specified in `granularitySpec`, Druid doesn't replace any segments even if `dropExisting` is `true`. WARNING: this feature is still experimental.|false|no|
```

**Review by 317brian — APPROVED** [review]
One minor nit, but otherwise LGTM. Thanks for updating the language and consistency

File: docs/ingestion/native-batch.md (hunk 1 above)
Comment (ektravel) [reply docs/ingestion/native-batch.md]: Good catch! Updated.

**Comment by 317brian** [comment]
Thanks!

**Pull request merged**